funcx_sdk/funcx/serialize/facade.py

```diff
@@ -49,93 +23,33 @@ def __init__(self, use_offprocess_checker=False):
-                for line in f.readlines():
-                    if "BINDING TO" in line:
-                        _, port = line.strip().split(":")
-                        port = int(port)
-                        return port
-                    elif "OFF_PROCESS_CHECKER FAILURE" in line:
-                        raise Exception(line)
-
-        raise Exception("Failed to determine off_process_checker's port")
-
     def _list_methods(self):
         return self.methods_for_code, self.methods_for_data
 
-    def cleanup(self):
-        logger.debug("Cleaning up")
-        if self.use_offprocess_checker:
-            self.off_proc_client.close()
-            self.serialize_proc.terminate()
-
-    def deserialize_check(self, serialized_msg, timeout=1):
-        """
-        Parameters
-        ----------
-        serialized_msg
-
-        timeout : int
-           timeout in seconds to wait for result
-        """
-        if not self.use_offprocess_checker:
-            return False
-
-        with self.serialize_lock:
-            status, info = self.off_proc_client.send_recv(serialized_msg)
-
-            if status == "SUCCESS":
-                return True
-            if status == "PONG":
-                return status
-            else:
-                raise Exception(info)
-
     def serialize(self, data):
         serialized = None
-        serialized_flag = False
         last_exception = None
 
         if callable(data):
-            for method in self.methods_for_code.values():
-                try:
-                    serialized = method.serialize(data)
-                    self.deserialize_check(serialized)
-                except Exception as e:
-                    logger.debug(
-                        f"[NON-CRITICAL] Method {method} failed with exception: {e}"
-                    )
-                    last_exception = e
-                    continue
-                else:
-                    serialized_flag = True
-                    break
-
+            stype, methods = "Callable", self.methods_for_code
         else:
-            for method in self.methods_for_data.values():
-                try:
-                    serialized = method.serialize(data)
-                except Exception as e:
-                    logger.exception(f"Method {method} did not work")
-                    last_exception = e
-                    continue
-                else:
-                    serialized_flag = True
-                    break
-
-        if serialized_flag is False:
+            stype, methods = "Data", self.methods_for_data
+        err_msg = f"{stype} Serialization Method {{}} failed with: {{}}"
+
+        for method in methods:
+            try:
+                serialized = method.serialize(data)
+                break
+            except Exception as e:
+                logger.info(err_msg.format(method, e))
+                last_exception = e
+                continue
```

Comment:
This is a nice cleanup, but the refactor has forgotten the iterable data structure.  I'm rather surprised the tests didn't break on this.  That is, when running locally, this fails with an `AttributeError`
```python
'str' object has no attribute 'serialize'
```
Which implies we have at least one new test to add as well!

Reply:
Oh jeez -- I need to learn to scroll.  The tests _did_ fail here.  [sigh]